internal/campaigns/executor.go

```diff
@@ -144,15 +144,19 @@ func (x *executor) do(ctx context.Context, task *Task) (err error) {
 		if result, err = x.cache.Get(ctx, cacheKey); err != nil {
 			err = errors.Wrapf(err, "checking cache for %q", task.Repository.Name)
 			return
-		} else if result != nil {
+		} else if result != nil && len(result.Commits) == 1 {
```

Comment:
Why this check?

I think it's here because you want to avoid that the `result.Commits[0].Diff` further down crashes, right? If that is the case, I'd move it there:

```go
var diff string
if len(result.Commits) >= 1 {
  diff = result.Commits[0].Diff
}

spec := createChangesetSpec(task, diff)
```

Otherwise, I fear, the "is something in the cache?"-check might be dependent on some side-effects (what if we _want_ to cache a result without any commits? etc.

Reply:
I was actually more concerned about the case where there are more than one commit. Is it fine to ignore the diffs of all subsequent commits?

I know that there is only one commit at the moment, but for future-proofing, I'd prefer a solution that either gracefully degrades or instantly fails. By calling it a cache miss, I was opting for "gracefully degrades". (Maybe I'm worrying about something I don't need to be worrying about?)

Reply:
(To be clear, I'm happy to change this for the zero-commits case as you suggest. I'm just not sure what to do when there are 2+ commits.)

Reply:
> I was actually more concerned about the case where there are more than one commit. Is it fine to ignore the diffs of all subsequent commits?

Right now, yeah. If we have more than one commit, _right now_, we should either panic because something's wrong or we can ignore it.

In the future, I think what we'd need to do here is to create a commit for each diff that we produce in a repository.

Reply:
Alright, I'll open another PR that accepts 0 or 1 commits, and panics on more than 1.